.github/workflows/.github/workflows/ARC_llama_recipes_pytest_on_single_nvidia_gpu_node.yaml

```diff
@@ -0,0 +1,92 @@
+name: "[ARC][GPU] llama-recipes Pytest tests on Self-Hosted GPU k8s Nodes."
+on:
+  pull_request:
+    branches:    
+      - 'main'
+    paths:
+      - 'src/llama-recipes/configs/*.py'
+      - 'src/llama-recipes/utils/*.py'
+      - 'src/llama-recipes/*.py'
+
+  # triggers workflow manually for debugging purposes.      
+  workflow_dispatch:
+    inputs:
+      runner:
+        description: 'GHA Runner Scale Set label to run workflow on.'
+        required: true
+        default: gha-runner-scale-set-nvidia-single-gpu
```

Comment:
@maximgroshev I wonder if for the pytest we would need to fall back to GPU/ I guess as the first step we are targeting CPU for Pytests?

Is this a reserved instance?

Reply:
I agree on running Pytest test on CPU k8s nodes. Currently there are still couple of dependencies on cuda in existing tests that make them fail on CPUI runners: https://github.com/maximgroshev/llama-recipes/actions/runs/7903261159